### PR TITLE
feat(COR-963): Helm - allow custom domain setup

### PR DESCRIPTION
**Pull request by pggb25**

### CODE REVIEW

File: qovery/data_source_helm.go [hunk 1]
```diff
@@ -470,6 +470,34 @@ func (d helmDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, re
...
+						"domain": schema.StringAttribute{
+							Description: "Your custom domain.",
+							Computed:    true,
+						},
+						"validation_domain": schema.StringAttribute{
+							Description: "URL provided by Qovery. You must create a CNAME on your DNS provider using that URL.",
+							Computed:    true,
+						},
+						"generate_certificate": schema.BoolAttribute{
+							Description: "Qovery will generate and manage the certificate for this domain.",
```
Comment (benjaminch) [inline qovery/data_source_helm.go:491]: _suggestion_:  `Qovery will generate and manage the certificate for this domain.` => `Qovery will generate and manage domain's certificate.`

File: qovery/data_source_helm.go [hunk 2]
```diff
@@ -470,6 +470,34 @@ func (d helmDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, re
...
+							Optional:    true,
+						},
+						"status": schema.StringAttribute{
+							Description: "Status of the custom domain.",
```
Comment (benjaminch) [inline qovery/data_source_helm.go:495]: _suggestion_: `Status of the custom domain.` => `Custom domain status.`

File: qovery/resource_helm.go [hunk 3]
```diff
@@ -454,6 +454,34 @@ func (r helmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
...
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Id of the custom domain.",
+							Computed:    true,
+						},
+						"domain": schema.StringAttribute{
+							Description: "Your custom domain.",
+							Required:    true,
+						},
+						"generate_certificate": schema.BoolAttribute{
```
Comment (benjaminch) [inline qovery/resource_helm.go:470]: _suggestion_: `Qovery will generate and manage the certificate for this domain.` => `Qovery will generate and manage domain's certificate.`

File: qovery/resource_helm.go [hunk 4]
```diff
@@ -454,6 +454,34 @@ func (r helmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
...
+							Description: "Qovery will generate and manage the certificate for this domain.",
+							Required:    true,
+						},
+						"validation_domain": schema.StringAttribute{
+							Description: "URL provided by Qovery. You must create a CNAME on your DNS provider using that URL.",
+							Computed:    true,
+						},
+						"status": schema.StringAttribute{
```
Comment (benjaminch) [inline qovery/resource_helm.go:478]: _suggestion_: `Status of the custom domain.` => `Custom domain status.`

File: qovery/data_source_helm.go [hunk 5]
```diff
@@ -470,6 +470,34 @@ func (d helmDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, re
 					},
 				},
 			},
+			"custom_domains": schema.SetNestedAttribute{
+				Description: "List of custom domains linked to this container.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Id of the custom domain.",
```
Comment (benjaminch) [inline qovery/data_source_helm.go:479]: _suggestion_: `Id of the custom domain.` => `Custom domain ID.`

File: qovery/data_source_helm.go [hunk 6]
```diff
@@ -470,6 +470,34 @@ func (d helmDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, re
...
+							Computed:    true,
+						},
+						"domain": schema.StringAttribute{
+							Description: "Your custom domain.",
```
Comment (benjaminch) [inline qovery/data_source_helm.go:483]: _suggestion_: `Your custom domain.` => `Domain name.`

File: qovery/resource_helm.go [hunk 7]
```diff
@@ -454,6 +454,34 @@ func (r helmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
...
+			"custom_domains": schema.SetNestedAttribute{
+				Description: "List of custom domains linked to this helm.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Id of the custom domain.",
+							Computed:    true,
+						},
+						"domain": schema.StringAttribute{
```
Comment (benjaminch) [inline qovery/resource_helm.go:466]: _suggestion_: `Your custom domain.` => `Domain name.`

File: qovery/resource_helm.go [hunk 8]
```diff
@@ -454,6 +454,34 @@ func (r helmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
 					},
 				},
 			},
+			"custom_domains": schema.SetNestedAttribute{
+				Description: "List of custom domains linked to this helm.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Id of the custom domain.",
```
Comment (benjaminch) [inline qovery/resource_helm.go:463]: _suggestion_: `Id of the custom domain.` => `Custom domain ID.`

**Review by benjaminch — APPROVED** [review]
LGTM, added couple wording suggestion for fields descriptions.

File: qovery/data_source_helm.go (hunk 1 above)
Comment (pggb25) [reply qovery/data_source_helm.go:491]: I will fix your comment in a separate PR to also change the descriptions of the other services (app, container...)

**Pull request merged**